aodndata/moorings/products_handler.py

```diff
@@ -43,25 +43,31 @@ def __init__(self, *args, **kwargs):
         self.allowed_extensions = ['.json_manifest']
         self.product_site_code = None
         self.product_variables = None
+        self.input_file_collection = None
+        self.input_file_variables = None
+        self.excluded_files = defaultdict(set)
 
-    def preprocess(self):
-        """Collect available input files and create the products, adding them to the collection to be published."""
-
-        # Read the manifest file and extract key parameters for product
+    def _read_manifest(self):
+        """Read the manifest file and extract key parameters for product"""
         with open(self.input_file) as f:
             manifest = json.load(f)
+
```

Comment:
I would avoid try here and ask explicitly the keys - faster and cleaner.
```
mkeys = manifest.keys();
invalid_file = 'site_code' not in mkeys and 'variables' not in mkeys;
if invalid_file:
```

Reply:
That is changing from EAFP to LBYL, which is not idiomatic:

https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Exceptions
https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/

Reply:
IMO, EAFP is overused for simple cases such as the above.

You use more lines, refactoring to list comprehensions (or generators) takes more time later, etc.

TO call that more "idiomatic" is a thin line - Why I should fall back to exception handling if I know what I'm looking for? If it was a function call wrap I would agree, but it's a simple comparison in standard classes.